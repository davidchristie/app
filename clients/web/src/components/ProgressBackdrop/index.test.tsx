import { render, screen } from "../../testing";
import { ProgressBackdrop } from ".";

describe("ProgressBackdrop", () => {
  describe("open", () => {
    it("renders correctly", () => {
      const { container } = render(<ProgressBackdrop open />);
      const element = screen.getByTestId("ProgressBackdrop");
      expect(element).toBe(container.firstChild);
      expect(element).toMatchSnapshot();
    });
  });

  describe("closed", () => {
    it("renders correctly", () => {
      const { container } = render(<ProgressBackdrop open={false} />);
      const element = screen.getByTestId("ProgressBackdrop");
      expect(element).toBe(container.firstChild);
      expect(element).toMatchSnapshot();
    });
  });
});
