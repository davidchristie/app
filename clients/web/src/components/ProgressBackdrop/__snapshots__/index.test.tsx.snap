// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBackdrop closed renders correctly 1`] = `
<div
  aria-hidden="true"
  class="MuiBackdrop-root makeStyles-backdrop-2"
  data-testid="ProgressBackdrop"
  style="opacity: 0; visibility: hidden;"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 100px; height: 100px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;

exports[`ProgressBackdrop open renders correctly 1`] = `
<div
  aria-hidden="true"
  class="MuiBackdrop-root makeStyles-backdrop-1"
  data-testid="ProgressBackdrop"
  style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
>
  <div
    class="MuiCircularProgress-root MuiCircularProgress-indeterminate"
    role="progressbar"
    style="width: 100px; height: 100px;"
  >
    <svg
      class="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx="44"
        cy="44"
        fill="none"
        r="20.2"
        stroke-width="3.6"
      />
    </svg>
  </div>
</div>
`;
